scintillometry/tests/test_folding.py

```diff
@@ -0,0 +1,90 @@
+# Licensed under the GPLv3 - see LICENSE
+"""Full-package tests of pulse folding."""
+
+import pytest
+import numpy as np
+import astropy.units as u
+from astropy.time import Time
+
+from ..base import Task
+from ..generators import EmptyStreamGenerator
+from ..pulse_folding import TimeFold
+from ..functions import Square
+
+
+class TestFoldingBase:
+    def setup(self):
+        self.start_time = Time('2010-11-12T13:14:15')
+        self.sample_rate = 10. * u.kHz
+        self.shape = (6000, 2, 4)
+        self.nh = EmptyStreamGenerator(shape=self.shape,
+                                       start_time=self.start_time,
+                                       sample_rate=self.sample_rate,
+                                       samples_per_frame=200, dtype=np.float)
+        self.sh = Task(self.nh, self.pulse_simulate, samples_per_frame=1)
+        self.period_bin = 125
+        self.phase_bin = 50
+
+    def phase(self, t):
+        dt = (t - t[0]).to(u.s)
+        F0 = 1.0 / (self.period_bin / self.sh.sample_rate)
+        return F0 * dt
+
+    def pulse_simulate(self, fh, data):
+        idx = fh.tell()
+        data [:]= 0
+        data += (10 if idx % self.period_bin ==0 else 0.125)
```

Comment:
I think this still needs the `np.where` I suggested before.

Reply:
I changed the pulse simulation algorithm. Before I place the pulse according to a phase function. But this method I just place the pulses every 125 bins.